### ISSUE TITLE
Pandas `aggregate` return type is Series instead of Dataframe

### ISSUE THREAD

**Issue by tekumara** [microsoft/pylance-release#276]
## Environment data

-   Language Server version: Pylance language server 2020.8.2 (pyright fbe2eb4d) starting
-   OS and version: macOS

## Expected behaviour

Dataframe is returned from `aggregate`.

## Actual behaviour

Series is returned from `aggregate`:

```
Expression of type "Series[Dtype]" cannot be assigned to return type "DataFrame"
  "Series[Dtype]" is incompatible with "DataFrame"
```

## Code Snippet / Additional information

```python
from typing import List
import pandas as pd

def group_tx(tx_df: pd.DataFrame, grouping_fields: List[str]) -> pd.DataFrame:
    grouped = tx_df.groupby(grouping_fields)[['date', 'amount']].aggregate(lambda x: list(x))
    return grouped
```

**Comment by gramster** [comment]
Looking at our stubs, calling `groupby` on a `DataFrame` returns a DataFrameGroupBy object. That has 3 overloads for `aggregate` all of which return `DataFrame`. So this doesn't seem to be a stub issue.

**Comment by gramster** [comment]
I split this code up into:

```
def group_tx(tx_df: pd.DataFrame, grouping_fields: List[str]) -> pd.DataFrame:
    foo = tx_df.groupby(grouping_fields)
    bar = foo[['date', 'amount']]
    grouped = bar.aggregate(lambda x: list(x))
    return grouped
```

It looks like `bar` gets inferred as a `DataFrame`, but then the call to `aggregate` is ambiguous, as it could be either `Series` or `DataFrame` (the right-hand expression on the line that assigns to `grouped` is flagged as an error:

```
Type of "aggregate" is partially unknown
  Type of "aggregate" is "Overload[(func: (*args: Unknown, **kwargs: Unknown) -> Unknown | str, axis: str | int = ..., **kwargs: Unknown) -> Series[Dtype], (func: List[(*args: Unknown, **kwargs: Unknown) -> Unknown] | Dict[_str, (*args: Unknown, **kwargs: Unknown) -> Unknown], axis: str | int = ..., **kwargs: Unknown) -> DataFrame]"
```

For some reason that then gets turned into `Series` in the last line, when I would expect it to be a union of Series and DataFrame. That seems like a pyright issue (yes, we could probably expand our overloads even more to the point where maybe we could disambiguate, but I don't think that changes things; AFAICT the type here should be `Union[DataFrame, Series]`, not `Series`.

**Comment by jakebailey** [comment]
There's an internal bug on this one as well; there's some oddity that comes from inheriting a function then adding overloads. I'm wondering if the lack of `@overload` on the parent class's function confuses things.

**Comment by debonte** [comment]
@tekumara there have been a lot of Pandas stub changes recently. I'm not sure when exactly this was fixed, but it is working for me in Pylance 2022.4.0. Can you see if it's working for you too?

**Comment by tekumara** [comment]
Thank you @debonte I'm no longer getting this error, and instead I'm getting a different one. I'll raise a new issue for that.
